0000-template.md

```diff
@@ -1,102 +1,118 @@
 
-> - What parts of the design do you expect to resolve through the RFC process
->   before this gets merged?
-> - What parts of the design do you expect to resolve through the implementation
->   of this feature before stabilization?
-> - What related issues do you consider out of scope for this RFC that could be
->   addressed in the future independently of the solution that comes out of this
->   RFC?
+> Is there a way to address this use case with the current product? What are the
+> downsides of implementing this feature?
 
-# Future Possibilities
+### What changes are required to enable this change?
 
-> Think about what the natural extension and evolution of your proposal would be
-> and how it would affect CDK as whole. Try to use this section as a tool to more
-> fully consider all possible interactions with the project and ecosystem in your
-> proposal. Also consider how this fits into the roadmap for the project.
+> Briefly describe the high-level design approach for implementing this feature.
 >
-> This is a good place to "dump ideas", if they are out of scope for the RFC you
-> are writing but are otherwise related.
+> As appropriate, you can add an appendix with a more detailed design document.
 >
-> If you have tried and cannot think of any future possibilities, you may simply
-> state that you cannot think of anything.
+> This is a good place to reference a prototype or proof of concept, which is
+> highly recommended for most RFCs.
 
-# Implementation Plan
+### Is this a breaking change?
 
-INTENTIONALLY LEFT BLANK: an implementation plan will be added when the RFC is
-scheduled for implementation.
+> If the answer is no. Otherwise:
+>
+> Describe what ways did you consider to deliver this without breaking users?
+>
+> Make sure to include a `BREAKING CHANGE` section above with a a description of the breaking
+> changes and the migration path.
+
+### What alternative solutions did you consider?
+
+> Briefly describe alternative approaches that you considered.
+
+### What is the high level implementation plan?
 
-> The implementation plan should analyze all the tasks needed in order to
-> implement this RFC, and the planned order of implementation based on
-> dependencies and analysis of the critical path.
+> Describe your plan on how to deliver this feature from prototyping to GA.
+> Especially think about how to "bake" it in the open and get constant feedback
+> from users before you stabilize the APIs.
 >
-> Either add the plan here or add link which references a separate document
-> and/or a GitHub Project Board (and reference it here) which manages the
-> execution of your plan.
+> If you have a project board with your implementation plan, this is a good
+> place to link to it.
+
+### Are there any open issues that need to be addressed later?
+
+> Describe any major open issues that this RFC did not take into account. Each
+> issue should have a GitHub issue associated with it and listed here.
+
+## Appendix: ....
```

Comment:
I would add a few suggestive appendices, ex: 'changes required', 'alternatives considered', 'implementation plan' and maybe 'detailed customer experience'. 
These are all of course optional but would give the user a good template, and cover most RFCs.

Reply:
Added some example.

Reply:
Suggestion: Similar to the working backwards section, I would prefer if we added a few appendix examples.

Reply:
We can add them later. I don't have a concrete list in my head and generally I don't want to encourage over-documentation.